src/index.ts

```diff
@@ -191,4 +243,22 @@ function transformDatesHelper(data: { [key: string]: any }) {
 
 function isPlainObject(value: any) {
     return Object.prototype.toString.call(value) == '[object Object]' && value.constructor.name === 'Object';
+}
+
+export function setCacheTimeout(milliseconds: number) {
+    cacheTimeout = milliseconds;
+}
+
+export function setSerializedDocumentTransformer(transformerFunction: Function) {
+    serializedDocumentTransformer = transformerFunction;
+}
+
+export function getCachedDocumentSnapshotPromise(documentReference: DocumentReference): CachedDocumentSnapshotPromise {
```

Comment:
Good implementation of the idea. Love how this works. I just feel like this is starting to get big, we should have more than one index and separate the logic by some common grounds, maybe cache fncs in a file,  includes fncs on another, and so on.